WordPress/src/main/java/org/wordpress/android/ui/publicize/PublicizeListViewModel.kt

```diff
@@ -15,25 +22,86 @@ class PublicizeListViewModel @Inject constructor(
     private val eventBusWrapper: EventBusWrapper,
     @Named(BG_THREAD) private val bgDispatcher: CoroutineDispatcher,
 ) : ScopedViewModel(bgDispatcher) {
+    private val _uiState = MutableLiveData<UIState>()
+    val uiState: LiveData<UIState> = _uiState
+
+    private val _actionEvents = SingleLiveEvent<ActionEvent>()
+    val actionEvents: LiveData<ActionEvent> = _actionEvents
+
+    private var shouldUpdateServices = true
+    private var twitterPublicizeService: PublicizeService? = null
+
     fun onSiteAvailable(siteModel: SiteModel) {
+        if (!shouldUpdateServices) {
+            return
+        }
+        shouldUpdateServices = false
         val siteId = siteModel.siteId
         updateServices(siteId)
-        updateConnections(siteId)
+    }
+
+    fun onTwitterDeprecationNoticeItemClick() {
+        twitterPublicizeService?.let {
+            _actionEvents.value = ActionEvent.OpenServiceDetails(it)
+        }
     }
 
     private fun updateServices(siteId: Long) {
```

Comment:
🔍  Nitpick: maybe it would make sense to update this name to something more generic like `updateList` since it now updates the `services` and `connections`. But in the end, everything is shown in the list as "services" so I am adding this as a "nitpick" as it actually might make sense to still call it `updateServices`. Thoughts?

Reply:
Makes sense to me. IMHO calling it `updateServices` is hiding the behavior of also updating the connections. Changing the name to `updateList`.